### PR TITLE
Errors wallet

### PR DESCRIPTION
**Pull request by gasiortomasz**
## Checklist
- [x] Change is small and easy to review (please split big changes into multiple PRs)
- [x] Change is consistent with architecture
- [x] Change is consistent with test architecture
- [x] Change is consistent with naming conventions
- [x] New code is covered with tests
- [x] Tests related to old code are updated
- [x] Documentation is up to date with changes

### CODE REVIEW

File: universal-login-wallet/src/services/TransferService.ts [hunk 1]
```diff
@@ -22,7 +23,7 @@ class TransferService {
         return this.transferTokens(transferDetails);
       }
     }
-    throw new Error('No user wallet provided!');
+    throw new UserWalletNotFound();
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/services/TransferService.ts]: You can use `ensure` instead: `ensure(this.walletService.userWallet, UserWalletNotFound)`

File: universal-login-wallet/src/services/WalletService.ts [hunk 2]
```diff
@@ -32,13 +33,13 @@ export default class WalletService {
   }
 
   setFutureWallet(userWallet: FutureWallet) {
-    ensure(this.state === 'None', Error, 'Wallet cannot be overrided');
+    ensure( this.state === 'None', WalletOverriden);
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/services/WalletService.ts]: :+1

File: universal-login-wallet/test/unit/services/WalletService.ts [hunk 3]
```diff
@@ -51,4 +51,13 @@ describe('WalletService', () => {
     expect(walletService.userWallet).to.deep.eq(userWallet);
     expect(walletService.state).to.be.eq('Deployed');
   });
+
+  it('should throw if future wallet is not set', () => {
+    expect(() => walletService.setDeployed(userWallet.name)).throws('Future wallet was not set');
```
Comment (marekkirejczyk) [inline universal-login-wallet/test/unit/services/WalletService.ts]: `expect(...).to.throw(...)`

File: universal-login-wallet/src/services/TransferService.ts (hunk 1 above)
Comment (gasiortomasz) [reply universal-login-wallet/src/services/TransferService.ts]: But in such case I would need to cast it explicitly to boolean.

Comment (gasiortomasz) [reply universal-login-wallet/src/services/TransferService.ts]: ensureNotNull :) nevermind

**Pull request merged**